crates/typst/src/math/equation.rs

```diff
@@ -223,7 +244,74 @@ impl Packed<EquationElem> {
         } else {
             vec![MathParItem::Frame(run.into_fragment(&ctx, styles).into_frame())]
         };
+        let mut pos_and_sizes: Vec<(Point, Size)> = Vec::new();
+        let mut x = Abs::zero();
 
+        // helper function to determine the vertical offset for a frame from MathParItem
+        let get_vertical_shift_fn = |frame: &Frame| {
+            let font_size = scaled_font_size(&ctx, styles);
```

Comment:
It is a bit unfortunate that this calculation is duplicated below.

Reply:
Updated in the newest commit. Now the function is called once and stored in an arrary